### ISSUE TITLE
Lengthy worker ping delays and timeout isses

### ISSUE THREAD

**Issue by WayneKeenan** [buildbot/buildbot#3787]
Using latest dev, commit 52335d3e02893c5c7c4aabab4b1e4a04383e4473
I'm seeing builds take > 6 minutes to get pas the initial 'pinging worker' step and then sometimes they also error with:

```python
[Failure instance: Traceback (failure with no frames): &lt;class 'twisted.internet.error.ConnectionLost'&gt;: Connection to the other side was lost in a non-clean fashion.
```

**Comment by WayneKeenan** [comment]
I should add that the master and workers are all connected to the same LAN switch and this doesn't occur with the latest release on PyPi.

**Comment by tardyp** [comment]
> this doesn't occur with the latest release on PyPi.
you mean this *didn't* occur before you switched to master?

Or you made a downgrade, and couldn't reproduce, and again made a upgrade and the problem reproduced?

I tried to look at the history since 0.9.13, and maybe this PR #3747 could introduce this kind of problem, even if this is quite unlikely. Could try to revert that and retry?

**Comment by tardyp** [comment]
I created a revert branch for that suspicious PR. Meanwhile, I updated our metabuildbot to the latest mainline, to see if we reproduce.

**Comment by WayneKeenan** [comment]
I've just gone thru another downgrade to PyPi version (v1) and then an upgrade back to f2e3cb408781df2d0a45d721d4fc556d43e5da8f (v2)

The test env is 1 master and 2 workers (Pi, Mac)

I'm only triggered a builder that runs on the Pi worker.

v1 works ok for all builders on all workers.
v2 works when I only had 1 (Pi) worker setup.
v2 exhibits the problem after I added a second (Mac) worker, even when only starting a builder on the Pi worker.


What commit do you want me to try?

**Comment by WayneKeenan** [comment]
It's not always that precise/repeatable to reproduce.  
 I'm also noticing that the workers aren't reliably reconnecting if only the master is restarted. 
Also, sometimes forcing a build doesn't even get to the 'pinging worker' stage, the webpage only has 2 tabs: properties and debug.

**Comment by WayneKeenan** [comment]
tl;dr  reducing the worker keepalive to 30 seconds helped, why not reduce the default from 600 to 40?


For a production system where things are stable and changes to the BB config aren't rapid then I can see a longer keep alive would be fine.

For someone new to buildbot and/or someone in a tight cycle of BB config change/reconfg/retest then a small value is very helpful. For a newbie, a low value helps reduce *perceived* instability/lagginess.

Is there a reason not to change the default keep alive from 600 seconds to, say, 30 seconds?
